src/approval/ApprovalController.test.ts

```diff
@@ -463,11 +463,28 @@ describe('approval controller', () => {
       expect(result).toStrictEqual('success1');
       expect(deleteSpy.callCount).toStrictEqual(numDeletions);
     });
+  });
 
-    it('throws on unknown id', () => {
-      expect(() => approvalController.accept('foo')).toThrow(
-        getIdNotFoundError('foo'),
-      );
+  describe('resolve (idempotent tests)', () => {
+    let approvalController: ApprovalController;
+    let numDeletions: number;
+    let deleteSpy: sinon.SinonSpy;
+
+    beforeEach(() => {
+      approvalController = new ApprovalController({
+        messenger: getRestrictedMessenger(),
+        showApprovalRequest: sinon.spy(),
+      });
+      // TODO: Stop using private methods in tests
+      deleteSpy = sinon.spy(approvalController as any, '_delete');
+      numDeletions = 0;
+    });
+
+    // If already is already resolved the method should not throw error
```

Comment:
Sorry I'm not understanding what this means and the setup for the test isn't giving me clues. Would you mind explaining this further? What is already resolved? I think I'm thrown off because usually when you test for idempotence you just call something twice.